DomainContent/BongoBingo/bingoWheel.js

```diff
@@ -0,0 +1,272 @@
+            calledNumbers = JSON.parse(numbers[0]);
+            var i = 1;
+            var newNumber;
+            while (i < 16) {
+                newNumber = "B " + String(i);
+                if (calledNumbers.indexOf(newNumber) === -1) {
+                    bingoSquares.push(newNumber);
+                }
+                i++;
+            }
+            while (i < 31) {
+                newNumber = "I " + String(i);
+                if (calledNumbers.indexOf(newNumber) === -1) {
+                    bingoSquares.push(newNumber);
+                }
+                i++;
+            }
+            while (i < 46) {
+                newNumber = "N " + String(i);
+                if (calledNumbers.indexOf(newNumber) === -1) {
+                    bingoSquares.push(newNumber);
+                }
+                i++;
+            }
+            while (i < 61) {
+                newNumber = "G " + String(i);
+                if (calledNumbers.indexOf(newNumber) === -1) {
+                    bingoSquares.push(newNumber);
+                }
+                i++;
+            }
+            while (i < 76) {
+                newNumber = "O " + String(i);
+                if (calledNumbers.indexOf(newNumber) === -1) {
+                    bingoSquares.push(newNumber);
+                }
+                i++;
+            }
+            shuffle(bingoSquares);
+        },
+
+        getLights: function() {
+            Entities.findEntities(position, SEARCH_RADIUS).forEach(function(nearbyEntity) {
+                var name = Entities.getEntityProperties(nearbyEntity, 'name').name;
+                if (name.indexOf("Bingo Wall Light") !== -1) {
+                    var lightNumber = name.substring(17, name.length);
+                    bingoWallLights[lightNumber] = nearbyEntity;
+                }
+            });
+        },
+
+        reset: function() {
+            Entities.callEntityServerMethod(_this.entityID, 'clearCalledNumbers');
+            bingoWallLights.forEach(function(light) {
+                Entities.editEntity(light, { locked: false });
+                Entities.editEntity(light, { visible: false });
+                Entities.editEntity(light, { locked: true });
+            });
+            var position = Entities.getEntityProperties(_this.entityID, 'position').position;
+            var bingoNumberTexts = Entities.findEntitiesByName("Bingo Wheel Number", position, 1);
+            Entities.editEntity(bingoNumberTexts[0], {
+                locked: false
+            });
+            Entities.editEntity(bingoNumberTexts[0], {
+                text: "BINGO",
+                lineHeight: 1.1
+            });
+            Entities.editEntity(bingoNumberTexts[0], {
+                locked: true
+            });
+            if (_this.interval) {
+                Script.clearInterval(_this.interval);
+            }
+        },
+
+        mousePressOnEntity: function(entityID, mouseEvent) {
```

Comment:
It might be worth investigating if the wheel works with this functionality offloaded to a server script, so that when you click on the wheel, if you have permissions, you request the server to handle this. I'm not sure whether the ESS can modify velocity properties successfully, but I think that would make the wheel more robust.